gdrapi.c

```diff
@@ -381,9 +387,14 @@ int gdr_copy_from_bar(void *h_ptr, const void *bar_ptr, size_t size)
             memcpy_cached_store_sse(h_ptr, bar_ptr, size);
             break;
         }
+
         // fall through
         gdr_dbgc(1, "using plain implementation of gdr_copy_from_bar\n");
         memcpy(h_ptr, bar_ptr, size);
+
+        // note: fencing is not needed because plain stores are used
+        //_mm_sfence();
+
```

Comment:
i am wondering if this comment is relevant give these are loads from the BAR

Reply:
my rationale is:
- reading from a BAR does not require a write fence
- writing to host memory via memcpy() does not require a write fence, as I assume that even if memcpy() were implemented using NT stores ( compiler / libc detail ) a correct implementation would also include a trailing write fence
so the comment should really be:
"note: assuming the memcpy() implementation has all the required write fences